More CMake fixes

MaartenBent opened the pull request:
Some more CMake fixes of issues I encountered when trying it on Linux.

The GTK build was missing the opengl files, causing missing symbols when building the opengl samples.

Building `test_gui` failed because `-lmedia` was missing (because `wxUSE_MEDIACTRL 0`). I added checks for all libraries used in `wx_exe_link_libraries` so they are only added when enabled.

MaartenBent committed:
CMake: Add missing OpenGL files in GTK build

MaartenBent committed:
CMake: Only link executables to enabled libraries

TcT2k reviewed (changes requested):
You should add the missing files to build/files and run upmake afterwards so the changes won't be overwritten

MaartenBent commented:
Done, but running upmake didn't work as expected.

It only seems to work when the variables are already present in `files.cmake` (`OPENGL_GTK_SRC` and `OPENGL_GTK_HDR`). After adding them manually, upmake will add the files.

vadz commented:
Yes, upmake never adds new variables, it only updates the contents of the existing ones. It's supposed to do one thing and one thing only: allow to have the list of source files (and maybe headers) in one place and update all the other make/project files automatically and is, by design, too dumb to do anything else.

MaartenBent committed:
CMake: Fix library order in wxMSW build

uuid should be linked before oleacc, otherwise it causes multiple definition of `IID_IAccessible' (with MinGW64 gcc).
To simplify even more, specify all required libraries in wxTOOLKIT_LIBRARIES, in the same order as in the makefiles.
Add uxtheme library, it is required since wxUxThemeEngine wrapper has been removed.

MaartenBent committed:
CMake: Remove Unix-only from OnFatalException description

MaartenBent committed:
CMake: Fix building samples with MinGW64

Do not enable debugrpt and flash examples.
Define UNICODE when building on Windows. This allows the sdk_exe example to build, because it includes the windows headers directly.

MaartenBent commented:
Rebased and added `uxtheme` library.

vadz commented on build/cmake/functions.cmake:
```diff
@@ -166,6 +166,9 @@ function(wx_set_target_properties target_name is_base)
     endif()
 
     if(wxUSE_UNICODE)
+        if(WIN32)
```
Curious, is this really necessary? Defining just `_UNICODE` is supposed to be enough -- even if defining this one won't do any real harm neither.

MaartenBent replied:
Yes, the Windows SDK headers require `UNICODE`, the c++ headers require `_UNICODE`.
Normally wxWidgets will define both if one of the two is defined (in `platform.h`), and Visual Studio also adds them by default. But `sdk_exe` doesn't include any wxWidgets headers, and MinGW does not automatically define `_UNICODE`, hence this fix.

vadz reviewed (approved):
Looks good to me, thanks!

@TcT2k I'll merge this soon if you don't object.

TcT2k reviewed (approved):
Looks good to me
